meeshkan/server/commands.py

```diff
@@ -1,91 +1,114 @@
-import logging
 import os
+from pathlib import Path
 
 import click
-import tornado.ioloop
-from tornado.httpserver import HTTPServer
-from tornado.ioloop import IOLoop
-from tornado.web import Application
+import daemonocle
 
-from .utils.routing import PathRouting, HeaderRouting, Routing
-from .admin.views import RestMiddlewareView, RestMiddlewaresView, StorageView
+from meeshkan.server.server.server import MockServer
+from .proxy.proxy import RecordProxyRunner
+from .utils.routing import PathRouting, HeaderRouting
 from ..schemabuilder.update_mode import UpdateMode
-from .admin.views import StorageView
-from .proxy.proxy import RecordProxy
-from .server.callbacks import callback_manager
-from .server.response_matcher import ResponseMatcher
-from .server.views import MockServerView
-from .utils.data_callback import RequestLoggingCallback
-from .utils.routing import PathRouting, HeaderRouting, Routing
 
 LOG_CONFIG = os.path.join(os.path.dirname(__file__), 'logging.yaml')
-logger = logging.getLogger(__name__)
 
-def make_admin_app():
-    return Application([
-        (r'/admin/storage', StorageView),
-        (r'/admin/middleware/rest/pregen', RestMiddlewaresView),
-        (r'/admin/middleware/rest/pregen/(.+)', RestMiddlewareView),
-    ])
 
-def start_admin(port):
-    app = make_admin_app()
-    http_server = HTTPServer(app)
-    http_server.listen(port)
-    logger.info('Starting admin endpont on http://localhost:%s/admin', port)
+def add_options(options):
+    def _add_options(func):
```

Comment:
Does this need functools.wraps?

Reply:
Probably, yes, because click utilizes function names somehow. It works but it may become broken in the future.

Reply:
Turned out that we don't. The actual wrapper here is an option from click package.